sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapRfRExamples.cs

```diff
@@ -0,0 +1,195 @@
+                    _complexMarketDataApi.UpsertComplexMarketData(scope, upsertComplexMarketDataRequest);
+                ValidateComplexMarketDataUpsert(upsertComplexMarketDataResponse,
+                    upsertComplexMarketDataRequest.Count);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void GetAndValidatePortfolioCashFlows(
+            LusidInstrument instrument,
+            string scope, string portfolioCode,
+            string recipeCode,
+            string instrumentID)
+        {
+            var swap = (InterestRateSwap) instrument;
+            var cashflows = _transactionPortfoliosApi.GetPortfolioCashFlows(
+                scope: scope,
+                code: portfolioCode,
+                effectiveAt: TestDataUtilities.EffectiveAt.AddYears(3),
+                windowStart: swap.StartDate.AddDays(-3),
+                windowEnd: swap.MaturityDate.AddDays(3),
+                asAt:null,
+                filter:null,
+                recipeIdScope: scope,
+                recipeIdCode: recipeCode).Values;
+            
+            Assert.That(cashflows.Count, Is.GreaterThanOrEqualTo(1));
+        }
+
+        [LusidFeature("F5-23")]
+        [TestCase("SOFR")]
+        [TestCase("ESTR")]
+        [TestCase("SONIA")]
+        [TestCase("TONA")]
+        [TestCase("SARON")]
+        public void InterestRateSwapRfRCreationAndUpsertionExample(string type)
+        {
+            // CREATE an interest rate swap (that can then be upserted into LUSID)
+            var swap = InstrumentExamples.CreateExampleInterestRateSwap(type);
+            
+            // ASSERT that it was created
+            Assert.That(swap, Is.Not.Null);
+
+            // CAN NOW UPSERT TO LUSID
+            var uniqueId = swap.InstrumentType + Guid.NewGuid().ToString(); 
+            var instrumentsIds = new List<(LusidInstrument, string)>{(swap, uniqueId)};
+            var definitions = TestDataUtilities.BuildInstrumentUpsertRequest(instrumentsIds);
+            
+            var upsertResponse = _instrumentsApi.UpsertInstruments(definitions);
+            ValidateUpsertInstrumentResponse(upsertResponse);
+
+            // CAN NOW QUERY FROM LUSID
+            var getResponse = _instrumentsApi.GetInstruments("ClientInternal", new List<string> { uniqueId });
+            ValidateInstrumentResponse(getResponse, uniqueId);
+            
+            var retrieved = getResponse.Values.First().Value.InstrumentDefinition;
+            Assert.That(retrieved.InstrumentType == LusidInstrument.InstrumentTypeEnum.InterestRateSwap);
+            var roundTripSwap = retrieved as InterestRateSwap;
+            Assert.That(roundTripSwap, Is.Not.Null);
+            Assert.That(roundTripSwap.MaturityDate, Is.EqualTo(swap.MaturityDate));
+            Assert.That(roundTripSwap.StartDate, Is.EqualTo(swap.StartDate));
+            Assert.That(roundTripSwap.Legs.Count, Is.EqualTo(swap.Legs.Count));            
```

Comment:
Why not check some data that distinguishes the type of swap being upserted, like IndexName?

Reply:
Same comment for other examples - comes back to question of the purpose of this SDK code. Examples or actual tests.... My view is that they are examples rather than complete tests (tests should exist in Lusid rather than be caught here).

Reply:
Fair enough, I would argue then that this is already too many assertions, but it will be fine as is.